html_css/v2/main-course/intermediate_css/frameworks-preprocessors.md

```diff
@@ -0,0 +1,91 @@
+
+While the two frameworks mentioned above are heavy hitters that have been around for a while, you should be aware that there are quite a few others at this point, as evidenced by all of the lists of popular frameworks on the web. Two other extremely popular frameworks are [Tailwind](https://tailwindcss.com/) and [Bulma](https://bulma.io/). There are many more out there.
+
+### Disadvantages of Frameworks
+
+Frameworks are great for rapidly producing sites with interfaces that end users can easily interact with. However, once you've taken a tour through some of the more popular ones, like Bootstrap, you'll start noticing an awful lot of similarities between a lot of sites you encounter, due to similar use of frameworks. Another concern with frameworks is that too many new developers jump into learning frameworks too early in their education; the prospect of not having to keep practicing writing vanilla CSS is very tempting. As a result, these developers do not get enough CSS practice under their belts to solidify the fundamentals of that very important language.
+
+Additionally, the process of overriding a framework's styling or debugging style issues on your page becomes very difficult if you haven't really mastered CSS fundamentals. It is imperative to understand what a framework is doing "under the hood" so you are equipped to handle these issues later (and trust us, you will).
+
+For the reasons above, **we recommend that you continue to focus on the fundamentals of CSS as you continue through the curriculum.** This lesson is merely designed to teach you _about_ frameworks and their role.
+
+### Preprocessors Overview
+
+Preprocessors (aka precompilers) are languages that help you write CSS more easily. They  eliminate code repetition and provide all sorts of time-saving and code-saving features. Some of the standard preprocessors in use are [SASS](https://sass-lang.com/), [LESS](https://lesscss.org/)and [Stylus](https://stylus-lang.com/).
+
+Note: If you've been using Rails, you'll be familiar with SASS already since it's baked in.  If you're not using Rails, you may need to add some additional steps to your workflow to get it set up, but it's usually worth it.
+
+Some of the useful features of preprocessors include:
+
+1. Programming in CSS but with more options available to you.
+2. Including variables in your CSS (not an option in vanilla CSS).
+3. Including loops in your CSS (again not an option in vanilla CSS).
+4. Including if/else statement in your CSS (see the pattern here?).
+5. Allowing you to create reusable code.
+6. And a lot more features, depending on which preprocessor you use.
+
+As you can see, preprocessors are extremely helpful tools; however, there are some potential disadvantages to using them. See the reading below for one list of disadvantages.
```

Comment:
So here i would say something like:

> preprocessors _do_ have some unique and helpful tools, but many of their most helpful features have been implemented in vanilla CSS, so it might not be worth the overhead of learning one unless you think you really _need_ these features.

Reply:
Will use that.